lib/open3.rb

```diff
@@ -33,56 +33,124 @@
 
 module Open3
 
-  # Open stdin, stdout, and stderr streams and start external executable.
-  # In addition, a thread to wait for the started process is created.
-  # The thread has a pid method and a thread variable :pid which is the pid of
-  # the started process.
+  # :call-seq:
+  #   Open3.popen3([env, ] command_line, options = {}) -> [stdin, stdout, stderr, wait_thread]
+  #   Open3.popen3([env, ] exe_path, *args, options = {}) -> [stdin, stdout, stderr, wait_thread]
+  #   Open3.popen3([env, ] command_line, options = {}) {|stdin, stdout, stderr, wait_thread| ... } -> object
+  #   Open3.popen3([env, ] exe_path, *args, options = {}) {|stdin, stdout, stderr, wait_thread| ... } -> object
+  #
+  # Basically a wrapper for Process.spawn that:
+  #
+  # - Creates a child process, by calling Process.spawn with the given arguments.
+  # - Creates streams +stdin+, +stdout+, and +stderr+,
+  #   which are streams $stdin, $stdout, and $stderr in the child process.
```

Comment:
`$stdin`, `$stdout`, and `$stderr` are global variables. They can be redefined and it does not affect the result of `Open3.popen3`. For example:

```ruby
require "open3"
require "tempfile"

$stdin = Tempfile.new
$stdout = Tempfile.new
$stderr = Tempfile.new

a = Open3.popen3("echo hi")
STDOUT.puts(a[1].read)
```

It's more accurate to say it captures the standard input, standard output, and standard error.

Reply:
Modified.